hardhat/contracts/WorldMap.sol

```diff
@@ -24,58 +27,48 @@ contract WorldMapCreator is IWorld {
         _planetFactory = new PlanetFactory();
     }
 
-    // function initialiseWorld () {
-        // TODO: consider using this function to call both
-        // _defineWorldMap and _addPlanets functions
-    // }
-
     function defineWorldMap(uint256 _worldIndex, uint256 _length, uint256 _breadth) public {
         // TODO: consider having the world index randomly generated
         require(existingWorlds[_worldIndex].Length == 0, "World already created with that index" );
-        
+
         WorldMap memory newWorldMap;
-        
+
         newWorldMap = WorldMap({
             worldIndex: _worldIndex,
             Length: _length,
             Breadth: _breadth
-            // PlanetsInWorld: 
+            // PlanetsInWorld:
         });
 
         existingWorlds[_worldIndex] = newWorldMap;
     }
 
-    function manualCreatePlanet(uint _planetID, uint _worldMapIndex,
-                uint _xCoord, uint _yCoord, uint _planetType) public {
+    function manualCreatePlanet(uint _worldMapIndex,
+                uint _xCoord, uint _yCoord, uint _planetType) public returns (uint) {
+        require(existingWorlds[_worldMapIndex].Length > 0, "world does not exist");
 
-        _planetFactory.createPlanet(_planetID, _worldMapIndex,
+        planetIndex += 1;
+
+        _planetFactory.createPlanet(planetIndex, _worldMapIndex,
                 _xCoord, _yCoord, _planetType);
         planetsInWorld[_worldMapIndex].push(planetIndex);
-        planetIndex += 1;
-    }
 
```

Comment:
Moving index here means indexing will start at 1 (which is fine)

Reply:
@chris-lovejoy since 0 is the default value, i think its a good practise to start at 1 to tell apart if a struct is just the default value=0 or the initialized value=1, 2, 3, 4